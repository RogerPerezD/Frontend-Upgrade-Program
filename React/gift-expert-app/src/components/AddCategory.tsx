import React,{ useState } from 'react';

interface Props{
    setCategories: React.Dispatch<React.SetStateAction<string[]>>;
}

const AddCategory: React.FC<Props> = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleChangeInput = ( event: React.FormEvent<HTMLInputElement>) =>{
        setInputValue( event.currentTarget.value );
    }

    const handleSubmit = async( event: React.FormEvent) =>{
        event.preventDefault();
        console.log('Sumit launch', inputValue);
        if (inputValue.trim().length > 2) {
            setCategories( (categories: string[]) => { 
                return [inputValue,...categories];
            });
        }
        setInputValue('');
    }


    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue } 
                onChange={ handleChangeInput }
            />
            <button type='submit'>Agregar</button>
        </form>
    );
}

export default AddCategory;
