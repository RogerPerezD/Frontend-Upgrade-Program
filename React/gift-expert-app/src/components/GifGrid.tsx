
// import React, { useEffect, useState } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

interface Props{
    category: string
}

export const GifGrid: React.FC<Props> = ({ category }) => { 
    
    const {data: images, loading} = useFetchGifs( {category} );
    return (
        <>
        <h3 className="animate__animated animate__bounce">{ category }</h3>
        { loading && <p className="animate__animated animate__flash">Loading</p>}
        <div className='card-grid'>
            <ol>
                {
                    images?.map( (elemet: {
                        id: string,
                        title: string,
                        url: string
                      }) =>{
                        return  <GifGridItem key={ elemet.id} { ...elemet}/>
                    })
                }
            </ol>
        </div>
        </>
    );
}
