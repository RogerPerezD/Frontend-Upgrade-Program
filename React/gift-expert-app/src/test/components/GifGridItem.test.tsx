import { mount, shallow, ShallowWrapper } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Test sobre GifGridItem', () => {

    let  wrapper: ShallowWrapper;
    let id: string; 
    let url: string; 
    let title: string; 


    beforeEach( () =>{
        id = '122';
        url = 'https://localhost/algo.jpg';
        title = 'naruto';
        wrapper = shallow( <GifGridItem id={ id } title={ title } url={ url }/>);
    });
    
    test('should show the component', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should has a paragraph whit the title', () => {
        const paragraph = wrapper.find('p');
        expect(paragraph.text()).toBe( title );
    });

    test('should element img contain a src and alt whit the title', () => {
        const img = wrapper.find('img');
        const src = img.prop('src');
        const alt = img.props().alt;

        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('should  have animate__bounce as a class', () => {
        const div = wrapper.find('div');
        const classDiv = div.props().className;
        const isClassInDiv = classDiv?.includes('animate__bounce');

        expect( isClassInDiv ).toBe( true );
    });

});