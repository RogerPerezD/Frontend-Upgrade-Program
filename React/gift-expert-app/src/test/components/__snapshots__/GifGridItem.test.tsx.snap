// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test sobre GifGridItem should show the component 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="naruto"
    src="https://localhost/algo.jpg"
  />
  <p>
    naruto
  </p>
</div>
`;
